### ISSUE TITLE
Large Finetuning training duration clarification

### ISSUE THREAD

**Issue by Nano1337** [Victorwz/Open-Qwen2VL#9]
In your paper, you only mention "SFT on LLaVA-665k instructions takes 48 A100-40G GPU hours" but the large finetune uses the Mammoth-10M dataset. 

However, it's unclear to me how long you train for. Given a 10M sample dataset with a batch size of 128, this would be 78k steps, which would take a very long time (>4 days) on a cluster of 8 GPUs. Is this correct?

**Comment by Victorwz** [comment]
Yes, it is correct.

**Comment by Nano1337** [comment]
I've implemented a version of on-the-fly sequence packing for stage 1 pretraining given the WDS datasets that I can stream from s3. 

Some challenges with implementation: 

1. I've been having some issues with cyclic/periodic losses, which I mostly reduced the effect of with multiple layers of shuffling (both tar paths and samples)
2. On-the-fly means that we will have to deal with suboptimal packing. However, my implementation usually achieves 94-95% of optimal so shouldn't be too big of a problem. 

A few follow-up questions: 

1. How much of a speedup is using sequence packing as compared to without? 
2. The large local SFT Mammoth dataset is undesirable as I'd prefer streaming. How much work would it be to convert the local dataset to a WDS version?

**Comment by Victorwz** [comment]
Do you want to chat about more details on wechat? My wechat id is lsjvictor98.

1. The sequence packing accelerates the pre-training for at least 2-3x.
2. I think with multiprocessing, it is not hard to convert the local dataset to WDS version. The largest issue is that you have to unzip all images zip files, which will take 1-2 days. After that, you can set up 8 processes to load the dataset and do the converting, which will also take 2-3 days.

**Comment by Nano1337** [comment]
Sent over a wechat friend request to chat more. I've already unzipped all image files and it took a few hours since I have a large devbox. Would it be possible for you to draft up a conversion script that I could run on my side into the expected format?
